lib/Onyx.js

```diff
@@ -718,6 +719,19 @@ function mergeCollection(collectionKey, collection) {
                 promises.push(AsyncStorage.multiSet(keyValuePairsForNewCollection));
             }
 
+            // If the flag is set, clear any removed keys in the collection
+            if (clearRemoved) {
```

Comment:
I'm not sure I agree with adding this functionality to `mergeCollection()`. 

In Onyx, "merge" is supposed to merge a new value (usually an object) into an older one. If the new value is missing a key that the old value has I wouldn't expect that it would be removed at all (even with the optional flag). So it feels like `mergeCollection()` should follow that same pattern.

If we need to remove a key from Onyx we should maybe just allow explicitly removing a key instead of having it happen in the `mergeCollection()`? Looks like `Onyx.remove()` is not a public method, but I'm not sure why it shouldn't be?

Could that work ?

Reply:
This is a good point, I've cleared my approval because I agree that this isnt something `merge` should be doing

Reply:
Yep, I did address this in my first comment. My reasoning behind putting it here was to avoid the alternative, which was creating a new method (setCollection) that would basically first call `mergeCollection`, and then, again, get AllKeys and do the new functionality. Would that work?

>If we need to remove a key from Onyx we should maybe just allow explicitly removing a key instead of having it happen in the mergeCollection()? Looks like Onyx.remove() is not a public method, but I'm not sure why it shouldn't be?

Removing the key is setting it to null, right? That is basically what the new code does for all keys in a collection that are not in the "new" collection passed. I think it is a useful method to have for a number of things. Basically, any collection where we can remove elements in the backend could benefit from this.
We can totally replicate the functionality by manually deleting the keys in each case, but why would we?

Reply:
I've moved the logic to a new function: `setCollection`. LMK what you think